### ISSUE TITLE
Can't understand how fees are working

### ISSUE THREAD

**Issue by kerohero** [bitpay/copay#4803]
Well, Today my friend send me 0.0001 BTC (0.06 USD). Receive was successfully. I tried to send them back to him. The wallet told me that the fee is 0.000155 BTC and that's bigger than all i have in the wallet. So how this work?

**Comment by KeyJockey** [comment]
Bitcoin transactions require a "miner fee" to process the transaction... the "correct" fee you have to pay depends on the current network load (how busy things are, and/or how fast you want to try and get your transaction confirmed, etc).  

The wallet software queries the network and (usually) calculates the "best" fee for the transaction depending on how *large* the particular transaction actually is (and this is "large" in terms of data size, not amount of money... bitcoin transactions don't care about the money value, it's all just sending data... so it's possible to send a million dollars worth for the same cost as sending a few cents worth).

Bottom line, a typical transaction these days may cost something like ฿0.0003 to ฿0.0005 bitcoin which is currently about 20-30 cents... but your friend only sent you ฿0.0001 which is worth about 6 cents.  He should have sent you at least a dollar's worth (฿0.0015) so you could actually DO something with it...

**Comment by kerohero** [comment]
Thanks for explaining this

**Comment by KeyJockey** [comment]
No problem... happy to help a new user.  Also I should add that the FEE situation in bitcoin has become something of a "big issue" over the last couple years as it's tied in with the huge "blocksize issue".  

In the early days of bitcoin the fee was totally optional: you could send transactions with no fee if you wanted to and they'd likely still get confirmed (albeit slowly and still sometimes not at all... it was just the chance you took IF you wanted to try for a free transaction).

Later on, however, due to a lot of reasons this was changed to always require some minimum fee.  That "normal" fee used to be about ฿0.0001 and there are still a lot of wallets and websites etc that have this number hard-coded into them as the "normal" fee (but which is often now totally insufficient).

Copay has implemented the "dynamic fee calculation" system of most modern up to date wallets, to handle figuring out the RIGHT fee for you... but personally I would still like to see more advanced control available for users to set their fee manually.  Take a look at my thread here called **Advanced User Fee Control (formerly "Manual Override Dynamic Fee") # 3111** if you're interested in learning more :smile:

**Comment by kerohero** [comment]
I saw that before. When I go to the wallet Bitcoin network fee I saw "0.000633/kIB" What's kIB?

**Comment by KeyJockey** [comment]
The fee is calculated as amount of bitcoin per whatever kilobytes of data that the transaction size needs... transactions are bigger or smaller depending on the number of "inputs" (i.e. previous transactions) that the wallet has control over.

For example, I have one separate wallet where I continually send small amounts of "loose change" bitcoin, here and there from "leftover" amounts in other wallets... kind of like those large buckets of coins that people toss their change into at the end of the day, LOL.  Over time this accumulates to ten or twenty dollars worth, a few cents worth at a time, and when it builds up enough I empty it back to an active wallet so that it's real, useful spendable money again. 

THIS transaction, however, is ALWAYS very "expensive" to empty the loose-change bucket because it has DOZENS of very small inputs consolidating into one new larger output... so the necessary miner fee that's calculated for this "empty the loose change bucket" transaction is always a few dollars, rather than being "normal" at 20-30 cents.  

This is an extreme case, however, and most normal users won't often see something this far out LOL but it illustrates the point about how bitcoin miner fee calculations work :smile:

**Comment by kerohero** [comment]
Thanks for everything 😃 How can I contact you without github?

**Comment by KeyJockey** [comment]
Well, I have email of course but what do you need that we can't discuss here? 😜

**Comment by kerohero** [comment]
Well, A lot of things so I prefer chatting xD

**Comment by KeyJockey** [comment]
I don't use any chat app, sorry... anyway better just post here since answers may help others with the same questions :)

**Comment by kerohero** [comment]
Okay 👌

**Comment by dabura667** [comment]
https://gitter.im/bitpay/copay

This is a chatroom which uses your github account. And you can do private messages to people... if they have gitter than they will probably get a notification if you send them a message.

**Comment by kerohero** [comment]
Thanks dabura667 😃

**Comment by bitjson** [comment]
merging into https://github.com/bitpay/copay/issues/4657

**Comment by SteveFla78** [comment]
I know this is an old thread but I hope someone can express something to me. Why though do we get insufficiant funds here on my account when say I hold over $70 in my wallet but it won't let me move but $40 of it? This constantly happens to me on this service and so bad that I am actually so sick of having $30 to $40 always held hostage in my BTC wallet that I basically will never be able to touch due to this. Definitely not miner fees, i am a miner myself and keep up with costs of fees daily and its definitely not top up fees with the math adding in miner fees...just doesn't add up. Yet for 6 months now I have held this random amount I can't move to an exchange, nor can load on my card. Why is this? I have kept with updates with Bitpay and I have still have this stuck balance. All I get is an insufficient fund notice when clearly there is a decent balance to move around? Think this is the end for me with Bitpay, let that balance stay stuck and take my business elsewhere.

**Comment by kerohero** [comment]
@SteveFla78 You have the private key, right? Did you tried to use it with blockchain.info? If not, Then do it and try to transfer the stuck balance to another wallet (maybe another BitPay wallet) now transfer it from the other wallet to yours and see what happens  (Just an Idea I didn't experience that before..)

**Comment by SteveFla78** [comment]
Not a bad idea @kerohero I can attempt that, done that before with another coin but for the life of me didn't come a conclusion to try that idea out lol. Thank you for that, green light hit as soon as you said that, might work to pull out then free it from this odd insufficient issue I am seeing. Its almost as though the wallet has no clue its there, always around the same amount but keeps kind of growing over time. Started with $5 but now its reaching an average of $30 I can't touch so I got to sort this one out in case it keeps persisting at this rate. I actually work over at Visio in their marketing and I actually chatted the other day with my dev their about this happening, his idea was it was some issue in wallet so maybe blockchain.info can help me pinpoint it. Its been frustrating, also happening to my fiance on her computer behind me lol....not as major as mine so I wonder how common this is or if its something to do with us here at the house

**Comment by kerohero** [comment]
I'm a dev, Trust me shit happens. I don't think that there is anything to do with you at the house. It must be a problem between the wallet and blockchain. It's not a common thing this is the first time to me to hear about a such problem

**Comment by SteveFla78** [comment]
Good to meet you @kerohero well I tried the import feature over at blockchain.info, no go...pulled out the private key from my backup file fine but its showing a zero balance, which is what I think is the issue now lol. Something is lost here and not showing up correctly I gather. As of right now I moved about $70 fine out but it wouldn't let me touch 0.003092 Bitcoin of mine, around $17.81 ...its stating spending this will need significant network fees as a warning in my Bitpay wallet and no matter what I do it states I have insufficient funds going on whether I try to move it out to another wallet or if I try to load the card side of the wallet. I could understand say if it was $5 but $17 should be plenty to cover both load up fee and network, I tried to load just $5 and nothing, then down to $1 top up and still says I have insufficient fees but there is a balance.

I am not really too stressed about it to be dead honest. Its minor change to me but concerns me only because I suspect this isn't how your system here is designed so figured maybe I should make aware of this to you. I gather it shouldn't do this so I am calm now that this isn't some Bitpay issue of holding peoples funds for some reason. Just odd though, my wallet started this about 2 months ago, but small amounts but seems to be growing up. Last week it was around $29 something I couldn't load, less today so it bounces around on me. My girl too, she though is stuck with I think $5 or so not able to move out or load, we just figured that is due to fees but on me its always a higher one.

**Comment by SteveFla78** [comment]
Maybe i have ghost coins lol. And I should create my own chain for these and put high asking prices on them lol. jk, got to have fun with these crypto headaches, trust me I am deep in some issues going with Visio and our community lately so I know the struggles you guys have to deal with support and development.

**Comment by kerohero** [comment]
Maybe the transaction size of those 0.003092 is big so the fee maybe even more than 0.003092. KeyJockey explained that well above.. I had the same problem but it was with a 10K of Satoshi xD. But a zero balance on blockchain. Then why there is a balance on BitPay, This interesting! 😃, Try to see the transactions history maybe you don't have any balance but a zero one. If not, Then it should be a very huge transaction It's a very high fee. You can calculate the size of it then the fee of it. This step is hard to do but I can't imagine another way to specify the problem, You can open an issue maybe you'll get a better answer/support , Good luck 😃.

**Comment by kerohero** [comment]
Yeah I got some fun with crypto cur problems too but it was a while ago. It's always fun to figure it out and getting some experience.

**Comment by SteveFla78** [comment]
I kind of wonder if maybe this Bitcoin Cash tie in from the fork has something to do with this, I have had this start up around there but pinpointing exacts is probably going to be hard for me because I move funds so much that i may have missed this happening until way after it started. I mine, work in crypto and constantly trade so its so hard to tell what is what in my wallet lol. But I may dig in today just to make sure maybe its really zero and just didn't balance out for some wild reason. I did put a support ticket in, kind of came to the GIT though to skim through to see if its something I can fix on my end but I think my issue is a rare one lol.

I also use the Shift card, but I hate Coinbase so much and why I came to Bitpay. Only plus with Shift is lack of having to load, just takes your coins on the fly per transaction...but have to hold in Coinbase so that is my LTC small payout avenue lol. Be cool to see something similar in Bitpay, who I actually rather of any of the cards I tried. Still a young space for all of us, going to have its growing pains.

Anyways I really appreciate the help @kerohero ....stick with the deving my friend, always better to work in this industry than scrap that change on the exchanges lol...plus its more fun, less stress I think lol. I will tinker around and do some digging but its okay, says the funds are there, just a matter of figuring what is causing this one. Could be the TX and fee maybe, but I should be above it on this last little amount. I do have a bigger check coming soon so I may see how that loads with this just to see maybe if it will take that way.

**Comment by dabura667** [comment]
@SteveFla78 Could you open Settings > Your Wallet > More Options > Wallet Addresses > Once everything is loaded, scroll to the bottom and look at "Total Wallet Inputs" and "Low amount inputs"

Tell me what it says there.

Also, tell me what type of wallet it is (single sig? if not, tell me the m and n values of the m-of-n multisig)

I had a bunch of low value inputs, and I was able to group them up into one during times of low fees (I would send with 1 satoshi/byte) by using the "Custom Fee" option on the send confirmation screen.

Thanks!

**Comment by militant13** [comment]
Have $100 trapped in bitpay wallet and they won’t send transaction

**Comment by militant13** [comment]
Shows it’s sending but on Blockchain nothing’s moving

**Comment by Tuny1** [comment]
Hi! 
I have receive some satoshis on my Copay Wallet. And have a notification that says:
**Amount too low to spend (฿0,0087=u$s114.-) ...** 
(this amount in the transaction is too low compared to the fees of the bitcoin network, to spend these funds it is required to have an amount higher than the commission required by the Bitcoin network)
Is it normal? Which amount is suppoused enough? Thank you, ;-)

**Comment by SteveFla78** [comment]
dabura667 sorry for the late response, I followed your walk through to the Wallet Addresses, all I see at the bottom is a bunch of addresses....and no Total Wallet Inputs or Low Amount Inputs. Maybe this is due to the recent update?

What I ended up figuring out though, if I use the Bitpay/Copay online Recovery Tool, and I let it scan from there, I can then find all the funds and move it out, which is seriously odd because the tool in the wallet never finds it. But if I extract my keys, go the JSON file route and use the online tool, it finds those lost stuck funds and it allows me to send out.

I think though this is a BTC issue going on, I have an Exodus wallet and that too is sticking on average about $17 unseen by the wallet. Yet to figure out how to pull that out but I think I might be able to use the same online recovery tool by Copay as long as I can get my keys out of that wallet. Also my Ledger Blue has the same ordeal all the sudden so this is some BTC craziness going on, less Bitpay now that I seen it happen in other areas of BTC wallets.

Not sure what is going on but I think I may contact Bitcoin Core, if they ever respond that is and see what they have to say about wallets not fully seeing complete amounts of funds. Weird situation but I did manage to get all out, cost me in miner fees of course but better than sitting there doing nothing lol.

**Comment by nosborbr** [comment]
dude Im stuck on bitpay as well. 
I have 0.006878 BTC in wallet and try send to my friend a amount of 0.00552451 BTC and it say insufficient fund. and it charge me 67.32% of the sending amount. is ridiculous more than half of the money. 
BTC/USD on app now  14.480,92.

**Comment by dabura667** [comment]
Import your phrase into Electrum.
